### DISCUSSION TITLE
bound instance methods

### DISCUSSION THREAD

**Discussion by ZeeD** [rokucommunity/brighterscript#681] [Ideas]
I was working on my project and I found that the methods are not bound to the instances. This means that in a snippet like

```brightscript
class Cls
    sub method()
        print m
    end sub
end class

obj = new Cls()
obj.method()

f = obj.method
f()
```

`m` will be different between the calls `obj.method()` and `f()`, with a behavior similar to what happen with `this` in javascript.

While, in retrospective, it was obvious looking at how the `.brs` is compiled, I would suggest to add some explicit note on the classes documentation (maybe it's me, but I didn't find any reference about it, and naively I expected to work with a more common scope resolution).

BTW: is there a way to force the value of `m`? something like the js `.bind()` function

**Comment by TwitchBronBron** [comment]
Sure, we can add a note in the class documentation. 

Unfortunately, the concept of `.bind()` doesn't exist in Roku development. Functions are just that. Functions. Any `this` binding happens by attaching the function to the object and then initiating the function using that dotted reference (i.e. `obj.method()`). As far as I'm aware, there's no way to fake this either.

**Reply by TwitchBronBron** [reply]
@ZeeD I just opened https://github.com/rokucommunity/brighterscript/pull/682 to address your concerns. Could you give it a quick read and approve it if that properly addresses your concerns?

**Reply by ZeeD** [reply]
yeah, I feared that there was no `.bind()` - and the only obvious way to implement it is by relying on closure scope, witch is not a thing on brightscript

**Reply by TwitchBronBron** [reply]
Right. No closures either. :(
